File: catalog/backends/java-spring/patterns/FeatureFlagBooleanGate.java
package com.example.service;

import com.datadoghq.openfeature.Provider;
import dev.openfeature.sdk.*;

// Pattern: Feature Flags — boolean gate with user targeting context
// Deps: add to pom.xml:
//   com.datadoghq:dd-openfeature:1.57.0
//   dev.openfeature:sdk:1.18.2
// Adapt: replace "checkout-new-flow" with your flag key; add domain attributes to context
public class FeatureFlagBooleanGate {

    private static final Client client;

    static {
        OpenFeatureAPI api = OpenFeatureAPI.getInstance();
        api.setProviderAndWait(new Provider());
        client = api.getClient();
    }

    public boolean isCheckoutEnabled(String userId, String plan) {
        EvaluationContext ctx = new MutableContext(userId)
            .add("plan", plan);
        return client.getBooleanValue("checkout-new-flow", false, ctx);
    }
}

File: catalog/backends/dotnet-aspnetcore/patterns/FeatureFlagBooleanGate.cs
using OpenFeature;
using OpenFeature.Model;
using Datadog.OpenFeature;

// Pattern: Feature Flags — boolean gate with user targeting context
// Deps: Datadog.Trace>=3.36.0 (.NET 6+), OpenFeature>=2.0.0
// Adapt: replace "checkout-new-flow" with your flag key; add domain attributes to context
public class FeatureFlagBooleanGate
{
    private static readonly FeatureClient _client;

    static FeatureFlagBooleanGate()
    {
        var provider = new DatadogProvider();
        Task.Run(() => Api.Instance.SetProviderAsync(provider)).GetAwaiter().GetResult();
        _client = Api.Instance.GetClient("service");
    }

    public async Task<bool> IsCheckoutEnabledAsync(string userId, string plan)
    {
        var ctx = EvaluationContext.Builder()
            .SetTargetingKey(userId)
            .Set("plan", plan)
            .Build();
        return await _client.GetBooleanValueAsync("checkout-new-flow", false, ctx);
    }
}
